firecrest/AsyncClient.py

```diff
@@ -843,10 +854,70 @@ async def compress(self, machine: str, source_path: str, target_path: str) -> st
             additional_headers={"X-Machine-Name": machine},
             data={"targetPath": target_path, "sourcePath": source_path},
         )
-        self._json_response([resp], 201)
+        timeout_str = "Command has finished with timeout signal"
+        if (
+            resp.status_code == 201 or
+            fail_on_timeout or
+            resp.status_code != 400 or
+            resp.json()["error"] != timeout_str
+        ):
+            self._json_response([resp], 201)
```

Comment:
Is this method not supposed to raise an exception in case fail_on_timeout is True and a timeout has occurred?
To improve code readability you could consider splitting this if, firs by checking the timeout case and then the others.

Reply:
Hm indeed I was not sure of what is the best order. The idea is:

- if the first request was successful then don't care for `fail_on_timeout`. We just return the response normally
- if the first request failed, but the user has `fail_on_timeout` we let `_json_response` take care of raising the appropriate error
- if the request error failed (but specifically with error code 400 and with the timeout error then we submit the job for compression. The reason that don't directly check the json is that in certain errors (like 500) we may not have a json to check.

Reply:
I will add at least a comment explaining this in the code.